langchain4j-vespa/src/test/java/dev/langchain4j/store/embedding/vespa/VespaEmbeddingStoreIT.java

```diff
@@ -0,0 +1,180 @@
+
+        TextSegment segment3 = TextSegment.from(
+            "But actually we tried our new swimming pool yesterday and it was awesome!"
+        );
+        Embedding embedding3 = embeddingModel.embed(segment3).content();
+
+        TextSegment segment4 = TextSegment.from("John Lennon was a very cool person.");
+        Embedding embedding4 = embeddingModel.embed(segment4).content();
+
+        embeddingStore.addAll(
+            asList(embedding1, embedding2, embedding3, embedding4),
+            asList(segment1, segment2, segment3, segment4)
+        );
+
+        // when
+        Embedding sportEmbedding = embeddingModel.embed("What is your favorite sport?").content();
+        List<EmbeddingMatch<TextSegment>> sportMatches = embeddingStore
+            .search(EmbeddingSearchRequest.builder().queryEmbedding(sportEmbedding).maxResults(2).build())
+            .matches();
+        Embedding musicEmbedding = embeddingModel.embed("And what about musicians?").content();
+        List<EmbeddingMatch<TextSegment>> musicMatches = embeddingStore
+            .search(EmbeddingSearchRequest.builder().queryEmbedding(musicEmbedding).maxResults(5).minScore(0.3).build())
+            .matches();
+
+        // then
+        assertThat(sportMatches).hasSize(2);
+        assertThat(sportMatches.get(0).score()).isCloseTo(0.639, percentage());
+        assertThat(sportMatches.get(0).embedded().text()).contains("football");
+        assertThat(sportMatches.get(1).score()).isCloseTo(0.232, percentage());
+        assertThat(sportMatches.get(1).embedded().text()).contains("swimming pool");
+
+        assertThat(musicMatches).hasSize(1);
+        assertThat(musicMatches.get(0).score()).isCloseTo(0.359, percentage());
+        assertThat(musicMatches.get(0).embedded().text()).contains("John Lennon");
+    }
+
+    @Override
+    protected EmbeddingStore<TextSegment> embeddingStore() {
+        return embeddingStore;
+    }
+
+    @Override
+    protected EmbeddingModel embeddingModel() {
+        return embeddingModel;
+    }
+
+    @Override
+    protected void clearStore() {
+        embeddingStore().removeAll();
+    }
+
+    @Override
+    protected void ensureStoreIsEmpty() {
+        await().atMost(Duration.ofSeconds(5)).untilAsserted(super::ensureStoreIsEmpty);
+    }
+
+    @Override
+    protected void ensureStoreIsReady() {
+        try {
+            getAllEmbeddings();
+        } catch (Exception e) {
+            deployVespaApp();
+        }
+    }
+
+    @Override
+    protected String documentURI(String id) {
+        return String.format("id:%s:%s::%s", DEFAULT_NAMESPACE, DEFAULT_DOCUMENT_TYPE, id);
+    }
+
+    @Override
+    protected Percentage percentage() {
+        return withPercentage(12);
+    }
+
+    @Override
+    protected List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        // we have to add records one by one, since Vespa's Json feeder in addAll method returns the ids in random order
```

Comment:
users are also affected? should we document this?

Reply:
Hmmm, the test relies on the proper sequence of IDs, I suppose for the users it's not relevant - but, yes, it makes sense to mention that point in the docs, looking into this...

Reply:
You mean that IDs returned in the `List<String>` from `addAll()` methods do not correspond (by order) to the embeddings that were added? This might be a big problem for some users that want to use these IDs to manage their embeddings later on...

Reply:
yes, the order of response IDs is random 😞 going to check the config options and possibly open an issue on Vespa side

Reply:
How big the performance impact would be? This only affects parsing, right? Or would it mean that there will be a network request for every embedding?

Reply:
Here good news as well - with #2106 the original IDs remain the same, they are not coming from the service any longer. I've refactored `EmbeddingStoreWithoutMetadataIT` back.